networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/gossip/subnets/AttestationSubnetSubscriptionsTest.java

```diff
@@ -73,16 +73,16 @@ void getChannelReturnsEmptyIfNotSubscribedToSubnet() {
     int subnetId = computeSubnetId(attestation);
     assertThat(computeSubnetId(attestation2)).isNotEqualTo(subnetId); // Sanity check
     subnetSubscriptions.subscribeToSubnetId(subnetId);
-    assertThat(subnetSubscriptions.getChannel(attestation)).isNotEqualTo(Optional.empty());
-    assertThat(subnetSubscriptions.getChannel(attestation2)).isEqualTo(Optional.empty());
+    assertThat(subnetSubscriptions.getChannel(attestation).join()).isNotEqualTo(Optional.empty());
+    assertThat(subnetSubscriptions.getChannel(attestation2).join()).isEqualTo(Optional.empty());
```

Comment:
These join calls are spreading through tests a lot.  It's not the end of the world but if for some reason the future wasn't completed they'd hang forever which would make it very hard to know which test was failing.

We already have `SafeFutureAssert` to add some utilities for asserting stuff about futures so I think we could add:
```suggestion
      public void isCompletedWithEmptyOptional() {
    isCompleted();
    Assertions.assertThat(actual.join()).isEqualTo(Optional.empty());
  }
  
  public void isCompletedWithNonEmptyOptional() {
    isCompleted();
    Assertions.assertThat(actual.join()).isNotEqualTo(Optional.empty());  
  }
```

Reply:
neat :tada: